src/InfiniteIterableValue.php

```diff
@@ -0,0 +1,315 @@
+<?php declare(strict_types=1);
+
+namespace GW\Value;
+
+final class InfiniteIterableValue implements IterableValue
+{
+    /** @var IterableValueIterator */
+    private $rootIterator;
+
+    /** @var object&callable */
+    private $iterator;
+
+    public function __construct(iterable $iterable)
+    {
+        $this->iterator = new IterableValueIterator($iterable);
+        $this->rootIterator = $this->iterator;
```

Comment:
Is that still needed? It's used only in `use` method, and it looks like that method is not working as it should be.

Reply:
Yes, its used in `use` method. Why do you think is not working properly? 

The `use` method is for swapping iterable passed to `IterableValue`, for example:

```php
$it = new InfiniteIterableValue([1, 2, 3]);
$it = $it->map(function (int $i): int { return $i + 1; });

var_dump($it->toArray()); // [2, 3, 4]
var_dump($it->use([10, 11, 12])->toArray()); // [11, 12, 13]
```

Reply:
`$this->rootIterator` is reference to first "iterator" that wraps original iterable passed to constructor. That allows to replace original iterable with something new, and in the same time keeping all decorators (filters, mappers).

And `$this->iterable` is decorated in several places, for example in `map`: https://github.com/gowork/values/pull/5/files#diff-25e183e677bbc88ec31e3380fc3d6ae8R104

Reply:
But does it mean that `use` mutates all `IterableValue` instances created from original iterator?

Reply:
Oh, I see..
So, we can remove `use` for now or try to implement it in different way

Reply:
Ok, I've change implementation :) Now `use` does not mutate other instances